src/StaticAnalyzers.cmake

```diff
@@ -3,8 +3,7 @@ macro(enable_cppcheck)
   if(CPPCHECK)
     set(CMAKE_CXX_CPPCHECK
         ${CPPCHECK}
-        --suppress=missingInclude
-        --enable=all
+        --enable=style
```

Comment:
I don't think we should limit the checks only to style. The default settings looked fine to me. Unless there is a bug that I haven't faced. It actually finds some errors that no other tool can find.

Reply:
Those settings didn't actually work. You cannot suppress missingInclude for all files on the command line, so that generated its own warnings.

According to the cppcheck manpage, `--style` enables all of the checks that we want.

![image](https://user-images.githubusercontent.com/234279/151566579-8bd75c30-4b69-4a04-a9f0-0776b0cc6d1d.png)

Reply:
I've explicitly added the other options, just in case the manpage is wrong.